src/BuildScriptGenerator/SdkStorageVersionProviderBase.cs

```diff
@@ -7,54 +7,58 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Net.Http;
-using System.Xml.Linq;
-using System.Xml.XPath;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
 using Microsoft.Oryx.Common;
 
 namespace Microsoft.Oryx.BuildScriptGenerator
 {
     public class SdkStorageVersionProviderBase
     {
         private readonly IEnvironment _environment;
-        private readonly IHttpClientFactory _httpClientFactory;
 
-        public SdkStorageVersionProviderBase(IEnvironment environment, IHttpClientFactory httpClientFactory)
+        public SdkStorageVersionProviderBase(IEnvironment environment)
         {
             _environment = environment;
-            _httpClientFactory = httpClientFactory;
         }
 
         protected PlatformVersionInfo GetAvailableVersionsFromStorage(
             string platformName,
             string versionMetadataElementName)
         {
-            var httpClient = _httpClientFactory.CreateClient("general");
-
             var sdkStorageBaseUrl = GetPlatformBinariesStorageBaseUrl();
-            var blobList = httpClient
-                .GetStringAsync($"{sdkStorageBaseUrl}/{platformName}?restype=container&comp=list&include=metadata")
-                .Result;
-            var xdoc = XDocument.Parse(blobList);
+            var blobServiceClient = new BlobServiceClient(serviceUri: new Uri(sdkStorageBaseUrl));
+            var blobContainerClient = blobServiceClient.GetBlobContainerClient(platformName);
+
+            // Example: nodejs-10.1.0.tar.gz is the format of a blob
+            var blobList = blobContainerClient.GetBlobs(BlobTraits.Metadata, prefix: $"{platformName}-");
             var supportedVersions = new List<string>();
-            foreach (var runtimeVersionElement in xdoc.XPathSelectElements(
```

Comment:
There seem to be no straight forward way to find elements by ignoring case, so I chose to use the Azure Blob API to read the blob metadata which is case-INsensitive. Due to this we no longer need HttpClient api.